File: RadialMenu/RadialMenuNet/Controller.cs
using System;
using SpaceVIL;
using SpaceVIL.Common;
using SpaceVIL.Core;

namespace RadialMenu
{
    public class Controller
    {
        private MainWindow _mainWindow;
        private Model _model;

        public Controller(Model model)
        {
            if (model == null)
                throw new ArgumentNullException("Model can not be NULL");
            AddStylesToDefaultTheme();
            _model = model;
            _mainWindow = new MainWindow();
        }

        private void AddStylesToDefaultTheme()
        {
            DefaultsService.GetDefaultTheme().AddDefaultCustomItemStyle(
                typeof(RadialMenu.RadialMenuItem),
                StyleFactory.GetRadialMenuDefaultStyle());

            DefaultsService.GetDefaultTheme().AddDefaultCustomItemStyle(
                typeof(RadialMenu.Contact),
                StyleFactory.GetContactStyle());
        }

        public void Start()
        {
            InitController();
            _mainWindow.Show();
        }

        private void InitController()
        {
            FillRadialMenu(_mainWindow.RadialMenu);

            _mainWindow.EventKeyPress += (sender, args) =>
            {
                if (args.Key == KeyCode.Menu)
                {
                    ShowContacts();
                }
            };

            _mainWindow.ShowContactsBtn.EventMouseClick += (sender, args) =>
            {
                ShowContacts();
            };

            _mainWindow.RadialMenu.AddButton.EventMouseClick += (sender, args) =>
            {
                _mainWindow.RadialMenu.Hide();
                InputDialog input = new InputDialog("Add a contact...", "Add");
                input.OnCloseDialog += () =>
                {
                    Contact contact = ItemFactory.GetContact(input.GetResult(), _mainWindow.Menu);
                    contact.SetNotificationCount(_model.GetRandomNumber(1, 19));
                    _mainWindow.RadialMenu.AddItem(contact);
                    _mainWindow.RadialMenu.Show();
                };
                input.Show(_mainWindow);
            };
        }

        private void FillRadialMenu(RadialMenuItem RadialMenu)
        {
            for (int i = 0; i < 8; i++)
                RadialMenu.AddItem(ItemFactory.GetContact(_model.getRandomName(), _mainWindow.Menu));

            foreach (Contact contact in RadialMenu.GetContacts())
            {
                contact.SetNotificationCount(_model.GetRandomNumber(1, 19));
            }
        }

        private void ShowContacts()
        {
            _mainWindow.RadialMenu.Show();
            _mainWindow.RadialMenu.SetFocus();
        }
    }
}

File: Apps/Demos/RadialMenu/RadialMenuJVM/src/main/java/RadialMenu/Controller.java
package RadialMenu;

import com.spvessel.spacevil.InputDialog;
import com.spvessel.spacevil.Common.DefaultsService;
import com.spvessel.spacevil.Flags.KeyCode;

public class Controller {
    private MainWindow _mainWindow;
    private Model _model;

    public Controller(Model model) {
        if (model == null)
            throw new NullPointerException("Model can not be NULL");
        addStylesToDefaultTheme();
        _model = model;
        _mainWindow = new MainWindow();
    }

    private void addStylesToDefaultTheme() {
        DefaultsService.getDefaultTheme().addDefaultCustomItemStyle(Contact.class, StyleFactory.getContactStyle());

        DefaultsService.getDefaultTheme().addDefaultCustomItemStyle(RadialMenuItem.class,
                StyleFactory.getRadialMenuDefaultStyle());
    }

    public void start() {
        initController();
        _mainWindow.show();
    }

    private void initController() {
        fillRadialMenu(_mainWindow.radialMenu);

        _mainWindow.eventKeyPress.add((sender, args) -> {
            if (args.key == KeyCode.MENU) {
                showContacts();
            }
        });

        _mainWindow.showContactsBtn.eventMouseClick.add((sender, args) -> {
            showContacts();
        });

        _mainWindow.radialMenu.addButton.eventMouseClick.add((sender, args) -> {
            _mainWindow.radialMenu.hide();
            InputDialog input = new InputDialog("Add a contact...", "Add");
            input.onCloseDialog.add(() -> {
                Contact contact = ItemFactory.getContact(input.getResult(), _mainWindow.menu);
                contact.setNotificationCount(_model.getRandomNumber(1, 19));
                _mainWindow.radialMenu.addItem(contact);
                _mainWindow.radialMenu.show();
            });
            input.show(_mainWindow);
        });
    }

    private void fillRadialMenu(RadialMenuItem radialMenu) {
        for (int i = 0; i < 8; i++)
            radialMenu.addItem(ItemFactory.getContact(_model.getRandomName(), _mainWindow.menu));

        for (Contact contact : radialMenu.getContacts()) {
            contact.setNotificationCount(_model.getRandomNumber(1, 19));
        }
    }

    private void showContacts() {
        _mainWindow.radialMenu.show();
        _mainWindow.radialMenu.setFocus();
    }
}